### PR TITLE
Update Migrate.swift

### PR DESCRIPTION
**Pull request by yasserislam**
## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have followed the [coding style](https://github.com/jessesquires/HowToContribute#style-guidelines), and reviewed the [contributing guidelines](https://github.com/jessesquires/HowToContribute).

## What's in this pull request?

nextModel always not equal destinationModel, so migration always fails..
when change equality to entityVersionHashesByName, it is working fine

### CODE REVIEW

**Comment by jessesquires** [comment]
Thanks @yasserislam ! 😄 

I'm not completely sure about this, but it looks like all tests pass. cc @marius-serban 

@yasserislam -- Can you write a unit test for this that fails before this patch, but passes after? 
(There are already some tests for the migration code.)

**Comment by marius-serban** [comment]
The change looks sound regardless, the `entityVersionHashesByNames` check is more explicit than `isEqual`. The same check should be performed on line 158 as well.
Don't know why `isEqual` fails though, it would be interesting to find out weather it's an iOS version thing or something else.

**Comment by jessesquires** [comment]
thanks @marius-serban !

File: Source/Migrate.swift [hunk 1]
```diff
@@ -145,7 +145,7 @@ internal func buildMigrationMappingSteps(bundle: Bundle,
         migrationSteps.append(nextStep)
         nextModel = nextStep.destination
 
-    } while nextModel != destinationModel
+    } while nextModel.entityVersionHashesByName != destinationModel.entityVersionHashesByName
```
Comment (jessesquires) [inline Source/Migrate.swift]: we'll need this same check at line 158

**Comment by jessesquires** [comment]
ping @yasserislam 😄

**Pull request merged**